bp_common/src/include/bp_common_aviary_defines.vh

```diff
@@ -207,18 +208,23 @@ typedef struct packed
                                                                                                    \
   , localparam dcache_sets_p              = proc_param_lp.dcache_sets                              \
   , localparam dcache_assoc_p             = proc_param_lp.dcache_assoc                             \
+  , localparam dcache_block_width_p       = proc_param_lp.dcache_block_width                       \
   , localparam icache_sets_p              = proc_param_lp.icache_sets                              \
   , localparam icache_assoc_p             = proc_param_lp.icache_assoc                             \
+  , localparam icache_block_width_p       = proc_param_lp.icache_block_width                       \
   , localparam acache_sets_p              = proc_param_lp.acache_sets                              \
   , localparam acache_assoc_p             = proc_param_lp.acache_assoc                             \
+  , localparam acache_block_width_p       = proc_param_lp.acache_block_width                       \
   , localparam lce_assoc_p                = `BSG_MAX(dcache_assoc_p,                               \
                                                      `BSG_MAX(icache_assoc_p, acache_assoc_p))     \
   , localparam lce_assoc_width_p          = `BSG_SAFE_CLOG2(lce_assoc_p)                           \
   , localparam lce_sets_p                 = `BSG_MAX(dcache_sets_p,                                \
                                                      `BSG_MAX(icache_sets_p, acache_sets_p))       \
   , localparam lce_sets_width_p           = `BSG_SAFE_CLOG2(lce_sets_p)                            \
                                                                                                    \
-  , localparam cce_block_width_p          = proc_param_lp.cce_block_width                          \
+  , localparam cce_block_width_p          = `BSG_MAX(dcache_block_width_p,                         \
```

Comment:
For now, there should be an assertion somewhere (nonsynth_if_verif?) that this is 512-bits (64-bytes), and that icache, dcache, and acache all have the same cache block width. I think allowing all the caches to use a different block size, such as 128 or 256-bits, should work (as long as they all have the same block size). Allowing different sizes per cache likely will not work without additional changes to the LCE/CCE.